lib/venice/client.rb

```diff
@@ -7,38 +7,31 @@ module Venice
   ITUNES_DEVELOPMENT_RECEIPT_VERIFICATION_ENDPOINT = 'https://sandbox.itunes.apple.com/verifyReceipt'
 
   class Client
-    attr_accessor :verification_url
-    attr_writer :shared_secret
-
     class << self
       def development
-        client = new
-        client.verification_url = ITUNES_DEVELOPMENT_RECEIPT_VERIFICATION_ENDPOINT
-        client
+        new(environment: :development)
       end
 
       def production
-        client = new
-        client.verification_url = ITUNES_PRODUCTION_RECEIPT_VERIFICATION_ENDPOINT
-        client
+        new(environment: :production)
       end
     end
 
-    def initialize
-      @verification_url = ENV['IAP_VERIFICATION_ENDPOINT']
+    def initialize(environment: :production, verification_url: ENV['IAP_VERIFICATION_ENDPOINT'])
```

Comment:
If an environment is provided, then the first `verification_url` found in this order will be used:
- `verification_url` option passed to the initialized
- `ENV['IAP_VERIFICATION_ENDPOINT']`
- `ITUNES_PRODUCTION_RECEIPT_VERIFICATION_ENDPOINT` or `ITUNES_DEVELOPMENT_RECEIPT_VERIFICATION_ENDPOINT` if `environment` matches `:production` or `:development`